src/compiler/args.rs

```diff
@@ -456,6 +458,17 @@ impl<T: ArgumentValue> ArgInfo<T> {
                     a => a?,
                 }
             }
+            ArgInfo::TakeArg(_s, create, ArgDisposition::ConcatenatedAndSeparated(_)) => {
+                if let Some(a) = get_next_arg() {
+                    Argument::WithValue(
+                        arg.to_string().leak(),
```

Comment:
it would have been nice to investigate this before submitting the PR ;)

Reply:
Believe me, I did. This is what I came up with in the end, and figured pushing a WIP so someone more knowledgeable in Rust could provide input. Would you rather I had given up on the patch and never submitted a PR instead because I couldn't come up with anything better? 🤷‍♂️

Reply:
Or did you mean before landing/pushing it into `master`? If so, yes, we definitely want to solve that first 😅